akka-persistence/src/main/scala/akka/persistence/state/javadsl/DurableStateStore.scala

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (C) 2009-2021 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.state.javadsl
+
+import java.util.Optional
+import java.util.concurrent.CompletionStage
+
+import scala.compat.java8.OptionConverters._
+
+import akka.annotation.ApiMayChange
+import akka.persistence.state.scaladsl.{ GetObjectResult => SGetObjectResult }
+
+/**
+ * API for reading durable state objects.
+ *
+ * For Scala API see [[akka.persistence.state.scaladsl.DurableStateStore]].
+ *
+ * See also [[DurableStateUpdateStore]]
+ *
+ * API May Change
+ */
+@ApiMayChange
+trait DurableStateStore[A] {
```

Comment:
What does the generic parameter stand for? Perhaps add that to the documentation section?

Reply:
Looking at GetObjectResult it's the type of the stored state - so this Store only stores objects of one type? Or are you allowed to pass `Any`/`Object`?

Reply:
It's the payload of the object. Fine with any `A` as long as you provide the serde for it. It's the same `A` that parameterizes the store. So if you define a `DurableStateStore[String]` then it has to be `GetObjectResult[String]`